content/articles/language-integrated-query-in-c#/index.md.md

```diff
@@ -0,0 +1,155 @@
+
+### Introduction
+The term "Language-Integrated Query" (LINQ) refers to a group of technologies that incorporate query capabilities directly into the C# programming language. Querying data has traditionally been plain text, with no type checking or IntelliSense assistance at build time. It's used to access data from objects, data sets, SQL Server, and XML, among other things.
+
+### Generating sequences with LINQ
+Make sure the below three lines are at the very top the Program.cs file produced by dotnet new console before you begin. For our program to run, the lines below should be in the program so that the results shall be obtained as expected

```

Comment:
Before this point, please explain how to create the application.

Reply:
@amosnjoro you're saying we add the three lines to program.cs file, how did we get to that program.cs file? Explain how to create the application and reach a point of having program.cs file.

Reply:
The first line of code which is using System means that you are using the System library in your project or program and that it  gives you some important functions and classes like WriteLine function and the Console class.

The second line of code which is using System.Collections.Generics  Contains interfaces and classes that define generic collections, which allow users to create strongly typed collections that provide better type safety and performance than non-generic strongly typed collections

The third line of code which is  using System.Linq provides classes and interfaces that support queries that use  (LINQ).

In short they act as preproccessor directives when dealing with LINQ. For example when connecting a window form to a database one have to include the line using System.Data.Sqlclient for it to work.

Reply:
@amosnjoro When someone reads your article, you start by saying we add the 3 lines into program.cs file. Assume the reader is a beginner, he/she will be wondering where that program.cs file is located.

Reply:
What I meant by doing so is that when one is writing a program in c# you have to include the using System line of code and when in the same program you go ahead and introduce a linq concept and that will prompt you to add the other lines of code so that the program may be perfect

Reply:
@amosnjoro please check this article [Azure functions in C#](https://www.section.io/engineering-education/using-azure-functions-and-csharp-to-create-serverless-apis/). It explains how to create the application before adding any block of code.